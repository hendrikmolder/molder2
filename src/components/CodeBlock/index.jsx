import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import prismStyles from '../../styles/prism'

const StyledContainer = styled.div`
    /* ${prismStyles}; */
    ${input =>
        console.log("CSS", input)}
`

const Title = styled.div`
    padding: 0.6rem 1rem;
    border-radius: 0.3em 0.3em 0 0;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    font-size: 14px;
    line-height: 1.375;
    background: #f3a953;
    color: #fff;
`

const CodeBlock = ({ input }) => {
    console.log(input)
    const title = input.primary.file_name

    return input.primary.file_name ?  (
        <React.Fragment>
            <Title>{title}</Title>
            <StyledContainer dangerouslySetInnerHTML={{ __html: input.primary.code_block.html }} />
        </React.Fragment>
    ) : <StyledContainer dangerouslySetInnerHTML={{ __html: input.primary.code_block.html }} />
}

CodeBlock.propTypes = {
    input: PropTypes.object.isRequired
}

export default CodeBlock