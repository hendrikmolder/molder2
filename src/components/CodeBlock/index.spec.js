import React from 'react'
import renderer from 'react-test-renderer'

import CodeBlock from './index'

describe('CodeBlock', () => {
    it('renders correctly with title', () => {
        const tree = renderer
            .create(<CodeBlock input={input}/>)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })


})



const codeBlock = `
<pre class="language-graphql">
                    <code class="language-graphql">
                        <span class="token keyword">query</span> MyQuery <span class="token punctuation">{</span>
  allPrismicPage <span class="token punctuation">{</span>
    edges <span class="token punctuation">{</span>
      node <span class="token punctuation">{</span>
        data <span class="token punctuation">{</span>
          body <span class="token punctuation">{</span>
            <span class="token operator">...</span> <span class="token keyword">on</span> <span class="token class-name">PrismicPageBodyCodeBlock</span> <span class="token punctuation">{</span>
              id
              primary <span class="token punctuation">{</span>
                code_block <span class="token punctuation">{</span>
                  html
                <span class="token punctuation">}</span>
                file_name
              <span class="token punctuation">}</span>
            <span class="token punctuation">}</span>
            <span class="token operator">...</span> <span class="token keyword">on</span> <span class="token class-name">PrismicPageBodyText</span> <span class="token punctuation">{</span>
              id
              primary <span class="token punctuation">{</span>
                text <span class="token punctuation">{</span>
                  html
                <span class="token punctuation">}</span>
              <span class="token punctuation">}</span>
            <span class="token punctuation">}</span>
          <span class="token punctuation">}</span>
        <span class="token punctuation">}</span>
      <span class="token punctuation">}</span>
    <span class="token punctuation">}</span>
  <span class="token punctuation">}</span>
<span class="token punctuation">}</span>
    </code>
</pre>`

const input = {
    primary: {
        code_block: codeBlock
    }
}