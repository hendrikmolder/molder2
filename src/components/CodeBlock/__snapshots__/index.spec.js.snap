// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeBlock renders correctly with title 1`] = `
<div
  className="sc-bdVaJa bqchxS"
>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": undefined,
      }
    }
  />
</div>
`;
